test/mock_job.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Mock Job class for test backends
+"""
+import uuid
+from qiskit.providers import BaseJob, JobStatus
+
+
+class MockJob(BaseJob):
+    """Mock Job class for tests"""
+
+    def __init__(self, backend, result):
+        self._result = result
+        super().__init__(backend, str(uuid.uuid4()))
+
+    def submit(self):
+        """Submit the job to the backend for execution."""
+        pass
+
+    def result(self):
+        """Return the results of the job."""
+        return self._result
+
+    def cancel(self):
+        """Attempt to cancel the job."""
+        pass
+
+    def status(self):
+        """Return the status of the job, among the values of ``JobStatus``."""
+        return JobStatus.DONE
```

Comment:
We currently have something almost identical in `test/mock_iq_backend.py` would be worth reusing what is already there. Note that refactoring the code to use the `MockJob` in this class would also be fine.

Reply:
I don't think this was merged yet when I added it. I'll refactor to use the one job class

Reply:
Thanks.